script.js

```diff
@@ -1,4 +1,139 @@
+var SCISSORS = "scissors";
+var PAPER = "paper";
+var STONE = "stone";
+var REVERSE_SCISSORS = "reverse scissors";
+var REVERSE_PAPER = "reverse paper";
+var REVERSE_STONE = "reverse stone";
+var numOfUserWins = 0;
+var numOfComputerWins = 0;
+var numOfDraws = 0;
+var numOfRounds = 0;
+var modeAskForUserName = "user types in their name";
+var modeStartPlaying = "sps game starts";
+var gameMode = modeAskForUserName;
+var userName = "";
+
+var programOutput = function () {
+  var randomDecimal = Math.random() * 3;
+  var randomInteger = Math.floor(randomDecimal);
+  var randomNumber = randomInteger + 1;
+  return randomNumber;
+};
+var computersRandomWord = function () {
+  var computersRandomNumber = programOutput();
+  if (computersRandomNumber == 1) {
+    return SCISSORS;
+  }
+  if (computersRandomNumber == 2) {
+    return PAPER;
+  }
+  if (computersRandomNumber == 3) {
+    return STONE;
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  // when the game starts, game mode is modeAskForUserName
+  if (gameMode == modeAskForUserName) {
+    userName = input;
+
+    // mode changes to sps game once the username is inputted.
+    gameMode = modeStartPlaying;
+    var myOutputValue = "Hi " + userName + "! Ready to play?";
+  } else if (gameMode == modeStartPlaying) {
+    var programChoice = computersRandomWord();
```

Comment:
This comment is related to my earlier points about using verbs when creating helper function names. Had we named the helper function with a verb in front (e.g. `getABall`), then we could assign the the "outcome" of this helper function to a variable that has a name that semantically makes sense. E.g. `var aBall= getABall()`. Or in your case, `var programChoice= getProgramChoice`.